R/maic_anchored.R

```diff
@@ -293,13 +321,178 @@ maic_anchored_tte <- function(res,
+    ci_l = NA,
+    ci_u = NA,
+    pval = NA
+  )
+
+  # : fit glm for binary outcome and robust estimate with weights
+  binobj_ipd <- glm(RESPONSE ~ ARM, ipd, family = glm_link)
+  binobj_ipd_adj <- glm(RESPONSE ~ ARM, ipd, weights = weights, family = glm_link)
+  binobj_agd <- glm(RESPONSE ~ ARM, agd, family = glm_link)
+
+  bin_robust_cov <- sandwich::vcovHC(binobj_ipd_adj, type = binary_robust_cov_type)
+  bin_robust_coef <- lmtest::coeftest(binobj_ipd_adj, vcov. = bin_robust_cov)
+  bin_robust_ci <- lmtest::coefci(binobj_ipd_adj, vcov. = bin_robust_cov)
+
+  res$inferential[["ipd_model_before"]] <- binobj_ipd
+  res$inferential[["ipd_model_after"]] <- binobj_ipd_adj
+  res$inferential[["agd_model"]] <- binobj_ipd
+
+  # derive ipd exp arm vs agd exp arm via bucher
+  res_AC <- res_template
+  res_AC$est <- bin_robust_coef[2, "Estimate"]
+  res_AC$se <- bin_robust_coef[2, "Std. Error"]
+  res_AC$ci_l <- bin_robust_ci[2, "2.5 %"]
+  res_AC$ci_u <- bin_robust_ci[2, "97.5 %"]
+  res_AC$pval <- bin_robust_coef[2, "Pr(>|z|)"]
+
+  if (is.null(res_BC)) res_BC <- as.list(summary(binobj_agd)$coefficients[c(2, 1:2)])
+  res_AB <- bucher(res_AC, res_BC, conf_lv = 0.95)
+
+  if (eff_measure %in% c("RR", "OR")) {
+    res_AB$est <- exp(mu)
+    res_AB$se <- sqrt((exp(sig^2) - 1) * exp(2 * mu + sig^2)) # log normal parameterization
+    res_AB$ci_l <- exp(res_AB$ci_l)
+    res_AB$ci_u <- exp(res_AB$ci_u)
+  } else if (eff_measure == "RD") {
+    res_AB$est <- mu * 100
+    res_AB$se <- sig * 100
+    res_AB$ci_l <- res_AB$ci_l * 100
+    res_AB$ci_u <- res_AB$ci_u * 100
+  }
+
+  # : get bootstrapped estimates if applicable
+  if (!is.null(weights_object$boot)) {
+    keep_rows <- setdiff(seq_len(nrow(weights_object$data)), weights_object$rows_with_missing)
+    boot_ipd_id <- weights_object$data[keep_rows, "USUBJID", drop = FALSE]
+
+    boot_ipd <- merge(boot_ipd_id, ipd, by = "USUBJID", all.x = TRUE)
+    if (nrow(boot_ipd) != nrow(boot_ipd_id)) stop("ipd has multiple observations for some patients")
+    boot_ipd <- boot_ipd[match(boot_ipd$USUBJID, boot_ipd_id$USUBJID), ]
+
+    stat_fun <- function(data, index, w_obj, eff_measure) {
+      r <- dynGet("r", ifnotfound = NA) # Get bootstrap iteration
+      if (!is.na(r)) {
+        if (!all(index == w_obj$boot[, 1, r])) stop("Bootstrap and weight indices don't match")
+        boot_ipd <- data[w_obj$boot[, 1, r], ]
+        boot_ipd$weights <- w_obj$boot[, 2, r]
+      }
+
+      boot_binobj_dat_adj <- glm(RESPONSE ~ ARM, boot_ipd, weights = boot_ipd$weights, family = glm_link)
+      boot_AC_est <- coef(boot_binobj_dat_adj)[2]
+      boot_AC_var <- vcov(boot_binobj_dat_adj)[2, 2]
+
+      boot_res_AC <- list(est = boot_AC_est, se = sqrt(boot_AC_var))
+      boot_res_AB <- bucher(boot_res_AC, res_BC, conf_lv = 0.95)
+
+      if (eff_measure %in% c("RR", "OR")) {
+        boot_AB_est <- exp(boot_AB_est)
+      } else if (eff_measure == "RD") {
+        boot_AB_est <- boot_AB_est * 100
+      }
```

Comment:
I think if we transform here we also need to take care about the se and variance estimates. I think they are used in the boot.ci calculation.

Reply:
Transform after bootstrap like in L390 in unanchored